src/simsopt/geo/curvehelical.py

```diff
@@ -1,69 +1,112 @@
-import jax.numpy as jnp
-from math import pi
 import numpy as np
+import jax.numpy as jnp
 from .curve import JaxCurve
 
-__all__ = ['CurveHelical']
-
-
-def jaxHelicalfouriercurve_pure(dofs, quadpoints, order, n0, l0, R0, r0):
-    A = dofs[:order]
-    B = dofs[order:]
-    phi = quadpoints*2*pi*l0
-    m, phiV = jnp.meshgrid(jnp.arange(order), phi)
-    AcosArray = jnp.sum(A*jnp.cos(m*phiV*n0/l0), axis=1)
-    BsinArray = jnp.sum(B*jnp.sin(m*phiV*n0/l0), axis=1)
-    eta = n0*phi/l0+AcosArray+BsinArray
-    gamma = jnp.zeros((len(quadpoints), 3))
-    gamma = gamma.at[:, 0].add((R0 + r0 * jnp.cos(eta)) * jnp.cos(phi))
-    gamma = gamma.at[:, 1].add((R0 + r0 * jnp.cos(eta)) * jnp.sin(phi))
-    gamma = gamma.at[:, 2].add(-r0 * jnp.sin(eta))
+__all__ = ["CurveHelical"]
+
+
+def jaxHelicalfouriercurve_pure(dofs, quadpoints, order, m, ell, R0, r):
+    """Pure function for the position vector used by CurveHelical."""
+    A = dofs[: order + 1]
+    B = jnp.concatenate(
+        (jnp.zeros(1), dofs[order + 1 :])
+    )  # Add 0 at the start for k = 0
+    phi = quadpoints * 2 * jnp.pi * ell
+    k, phi_2D = jnp.meshgrid(jnp.arange(order + 1), phi)
+    eta = m * phi / ell + jnp.sum(
+        A * jnp.cos(k * phi_2D * m / ell) + B * jnp.sin(k * phi_2D * m / ell), axis=1
+    )
+    R = R0 + r * jnp.cos(eta)
+    x = R * jnp.cos(phi)
+    y = R * jnp.sin(phi)
+    z = -r * jnp.sin(eta)
+    gamma = jnp.column_stack((x, y, z))
     return gamma
 
 
 class CurveHelical(JaxCurve):
-    r'''Curve representation of a helical coil.
-    The helical coil positions are specified by a poloidal angle eta that is a function of the toroidal angle phi with Fourier coefficients :math:`A_k` and :math:`B_k`.
-    The poloidal angle is represented as 
+    r"""Curve representation of a helical coil lying on a circular-cross-section
+    axisymmetric torus.
+
+    This curve type can describe the coils from the
+    `Hanson & Cary (1984) <https://doi.org/10.1063/1.864692>`__
+    and
+    `Cary & Hanson (1986) <https://doi.org/10.1063/1.865539>`__
+    papers on optimization for reduced stochasticity, as well as the Compact
+    Toroidal Hybrid (CTH) device at Auburn University.
+
+    The helical coil lies on a circular-cross-section axisymmetric torus with
+    major radius :math:`R_0` and minor radius :math:`r`. Following Hanson and
+    Cary, the position along the coil is written in terms of a poloidal angle
+    on the winding surface :math:`\eta` and the standard toroidal angle :math:`\phi`:
+
+    .. math::
+        \begin{align*}
+        x &= [R_0 + r \cos(\eta)] \cos(\phi), \\
```

Comment:
This is not exactly the formula for R and Z in (14a,b) of Stochasticity reduction (Cary Hanson) and it seems like the representation in that paper is more general than what is here -- if I've understood correctly can this be clarified in docstring?